src/classes/button.py

```diff
@@ -0,0 +1,30 @@
+import pygame
+
+class Button:
+
+    def __init__(self, screen: pygame.Surface, color: tuple[int, int, int], hover_color: tuple[int, int, int], start_point: tuple[int, int], dim: tuple[int, int], text: str, on_click=None, cargs=()) -> None:
+        self.screen = screen
+        self.color = color
+        self.hover_color = hover_color
+        self.x = start_point[0]
+        self.y = start_point[1]
+        self.width = dim[0]
+        self.height = dim[1]
+        self.text = text
+        self.on_click = on_click
+        self.cargs = cargs
+
+    def is_selected(self) -> bool:
+        mouse = pygame.mouse.get_pos()
+        return self.x <= mouse[0] <= self.x + self.width and self.y <= mouse[1] <= self.y + self.height
+
+    def click(self) -> None:
+        if self.is_selected() and self.on_click:
+            self.on_click(*self.cargs)
+
+    def draw(self) -> None:
```

Comment:
I started a functional approach (less OOP) so all render methods are in render.py

click and draw would be in this file but separate from the object.

- [ ] Let me know what you think, it does not have to be done this way

Reply:
I'll look into this, although I quite like the way the Button Class functions currently since it is all-encapsulated.

Reply:
I agree. there are some tradeoffs.

If we have all-encapsulated, then everything related to button/vehicle/etc is in their respective file. but every class has to worry about all functions (rendering, calculations, helpers)

if we have a page for each different function, then the classes only have to worry about 'object-ness' no functions. but then each file has to provide function for every class

class0:
  render()
  helper()
  calc()

class1:
  render()
  helper()
  calc()
...

vs

render:
  class0()
  class1()

helper:
  class0()
  class1()

calc:
  class0()
  class1()